engine/runtime/src/test/java/org/enso/compiler/CompilerTest.java

```diff
@@ -31,4 +37,82 @@ protected static IR.Module parse(String code) throws IOException {
     assertNotNull("IR was generated", ir);
     return ir;
   }
+
+  static void assertIR(String msg, IR.Module old, IR.Module now) throws IOException {
+    Function<IR, String> filter = f -> simplifyIR(f, true, true, false);
+    String ir1 = filter.apply(old);
+    String ir2 = filter.apply(now);
+    if (!ir1.equals(ir2)) {
+      String name = findTestMethodName();
+      Path home = new File(System.getProperty("user.home")).toPath();
+      Files.writeString(home.resolve(name + ".1"), ir1, StandardOpenOption.TRUNCATE_EXISTING, StandardOpenOption.CREATE, StandardOpenOption.WRITE);
+      Files.writeString(home.resolve(name + ".2"), ir2, StandardOpenOption.TRUNCATE_EXISTING, StandardOpenOption.CREATE, StandardOpenOption.WRITE);
+      assertEquals(msg, ir1, ir2);
+    }
+  }
+
+  private static String findTestMethodName() {
+    for (var e : new Exception().getStackTrace()) {
+      if (e.getMethodName().startsWith("test")) {
+        return e.getMethodName();
+      }
+    }
+    throw new IllegalStateException();
+  }
+
+  private static String simplifyIR(IR i, boolean noIds, boolean noLocations, boolean lessDocs) {
```

Comment:
One thing I'm missing is if you can describe what does the simplification do and when one should add additional cases (in case new IR is  added)?
I realize this code has been there before.